client/my-sites/checkout/checkout/source-payment-box.jsx

```diff
@@ -0,0 +1,189 @@
+import PaymentChatButton from './payment-chat-button';
+import config from 'config';
+import { PLAN_BUSINESS } from 'lib/plans/constants';
+import CartToggle from './cart-toggle';
+import AlternativePaymentMethods from './alternative-payment-methods';
+import TermsOfService from './terms-of-service';
+import Input from 'my-sites/domains/components/form/input';
+import cartValues from 'lib/cart-values';
+import SubscriptionText from './subscription-text';
+import analytics from 'lib/analytics';
+import wpcomUndocumented from 'lib/wpcom-undocumented';
+
+class SourcePaymentBox extends PureComponent {
+	static propTypes = {
+		paymentType: PropTypes.string.isRequired,
+		cart: PropTypes.object.isRequired,
+		transaction: PropTypes.object.isRequired,
+		paymentMethods: PropTypes.array.isRequired,
+		onSelectPaymentMethod: PropTypes.func.isRequired,
+	}
+
+	getLocationOrigin( l ) {
+		return l.protocol + '//' + l.hostname + ( l.port ? ':' + l.port : '' );
+	}
+
+	redirectToPayment( event ) {
+		const origin = this.getLocationOrigin( location );
+		event.preventDefault();
+
+		this.setSubmitState( {
+			info: translate( 'Sending details to %(paymentProvider)s', {
+				args: { paymentProvider: this.getPaymentProviderName() } } ),
+			disabled: true
+		} );
+
+		let cancelUrl = origin + '/checkout/';
+
+		if ( this.props.selectedSite ) {
+			cancelUrl += this.props.selectedSite.slug;
+		} else {
+			cancelUrl += 'no-site';
+		}
+
+		const dataForApi = assign( {}, this.state, {
+			successUrl: origin + this.props.redirectTo(),
+			type: this.props.paymentType,
+			cancelUrl,
+			cart: this.props.cart,
+			domainDetails: this.props.transaction.domainDetails
+		} );
+
+		// get thre redirect URL from rest endpoint
+		wpcomUndocumented.sourcePaymentUrl( dataForApi, function( error, sourcePaymentUrl ) {
+			let errorMessage;
+			if ( error ) {
+				if ( error.message ) {
+					errorMessage = error.message;
+				} else {
+					errorMessage = translate( 'Please specify a country and postal code.' );
+				}
+
+				this.setSubmitState( {
+					error: errorMessage,
+					disabled: false
+				} );
+			}
+
+			if ( sourcePaymentUrl ) {
+				this.setSubmitState( {
+					info: translate( 'Redirecting you to our payment provider' ),
+					disabled: true
+				} );
+				analytics.ga.recordEvent( 'Upgrades', 'Clicked Checkout With Source Payment Button' );
+				analytics.tracks.recordEvent( 'calypso_checkout_with_source_' + this.props.paymentType );
+				location.href = sourcePaymentUrl;
+			}
+		}.bind( this ) );
+	}
+
+	renderButtonText() {
+		if ( cartValues.cartItems.hasRenewalItem( this.props.cart ) ) {
+			return translate( 'Purchase %(price)s subscription with %(paymentProvider)s', {
+				args: { price: this.props.cart.total_cost_display, paymentProvider: this.getPaymentProviderName() },
+				context: 'Pay button on /checkout'
+			} );
+		}
+
+		return translate( 'Pay %(price)s with %(paymentProvider)s', {
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?